[flang] Disable noalias by default

tblah opened the pull request:
With these enabled we see a 70% performance regression for exchange2_r on neoverse-v1 (aws graviton 3) using `-mcpu=native -Ofast -flto`. There is also a smaller regression on neoverse-v2.

This appears to be because function specialization is no longer kicking in during LTO for digits_2. This can be seen in the output executable: previously it contained specialized copies of the function with names like `_QMbrute_forcePdigits_2.specialized.4`. Now there are no names like this.

The bug is not in flang - instead in the function specialization pass - but due to the size of the regression I would like to request that this is disabled until function specialization has been fixed.

vzakhari commented on flang/lib/Optimizer/Passes/Pipelines.cpp:
```diff
@@ -350,15 +350,11 @@ void createDefaultFIRCodeGenPassPipeline(mlir::PassManager &pm,
     framePointerKind = mlir::LLVM::framePointerKind::FramePointerKind::None;
 
-  bool setNoCapture = false, setNoAlias = false;
-  if (config.OptLevel.isOptimizingForSpeed())
```
Tom, will you be okay if I just remove this `if`?  The LIT tests should still work, and it will be easier to re-enable the code, when function speciailization is fixed.

tblah committed:
[flang] Disable noalias captures(none) by default

This is due to a 70% regression in exchange2_r on neoverse-v2
due to function specialization no longer triggering in the LTO
pipline.

tblah committed:
Re-enable captures(none)

vzakhari reviewed (approved):
LGTM. Thank you, Tom!
